### PR TITLE
LVNOD-54-Create-Redux-Setup

### PR DESCRIPTION
**Pull request by denisbodnar**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-54)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-54)

## Code reviewers

- [x] @yurii-kokhalevych
- [x] @AlbaMontis
- [x] @shyyka

### Second Level Review

- [x] @lhalam
- [x] @4asof4ik

### CODE REVIEW

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: app/src/components/Test/Test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+class Test extends React.Component {
+  constructor (props, context) {
+    super(props, context);
+
+    this.state = {
+      item: {title: ''}
+    };
+
+    this.onTitleChange = this.onTitleChange.bind(this);
```
Comment (AlbaMontis) [inline app/src/components/Test/Test.js]: With js6 arrow functions you don't need to bind

File: app/src/components/Test/Test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    this.onClickSave = this.onClickSave.bind(this);
+  }
+
+  onTitleChange(event) {
```
Comment (AlbaMontis) [inline app/src/components/Test/Test.js]: onTitleChange = (event) => {...

File: app/src/components/Test/Test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          type='submit'
+          value='Save'
+          onClick={this.onClickSave}
+        />
+      </div>
+    );
+  }
+}
+
+function mapStatetoProps(state, ownProps) {
```
Comment (AlbaMontis) [inline app/src/components/Test/Test.js]: Where do you use ownProps?

Comment (denisbodnar) [reply app/src/components/Test/Test.js]: ownProps represents any props passed to our component

https://stackoverflow.com/questions/41198842/what-is-the-use-of-the-ownprops-arg-in-mapstatetoprops-and-mapdispatchtoprops

**Review by AlbaMontis — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: app/src/reducer/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { combineReducers } from 'redux';
```
Comment (4asof4ik) [inline app/src/reducer/index.js]: Folder redux,  file name reducer

File: app/src/reducer/itemReducer.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export default function itemReducer(state = [], action) {
```
Comment (4asof4ik) [inline app/src/reducer/itemReducer.js]: File name example: container "Test" => reducer test.js and import into reducer.js , rootReducer

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Review by AlbaMontis — APPROVED** [review]

**Pull request merged**